### PR TITLE
Update indexing routes and osgi config

### LINKED ISSUE #93: Service document indexing does not work
Neither indexing nor reindexing results in any triples persisted to the triple store.  The [integration test](https://github.com/fcrepo4-labs/fcrepo-api-x/blob/master/fcrepo-api-x-integration/src/test/java/org/fcrepo/apix/integration/KarafServiceIndexingIT.java) is currently disabled due to dependency clashes that puncture the OSGi barrier between Jena 2 (from fcrepo-camel-toolbox) and Jena 3 (from api-x)

### PR DESCRIPTION
**Pull request by birkland**
Indexing routes were not filtering messages property, and were assuming
the presence of the wrong headers in reindexing messages; Verified manually

Resolves #93

### CODE REVIEW

**Comment by acoburn** [comment]
Just by way of FYI, fcrepo-camel/4.5.0 will not have the `JmsHeaders` class. Instead, camel users will be encouraged to follow this pattern:

    from("broker:queue:foo")
      .process(new EventProcessor())
      .filter(header(FCREPO_URI).contains("..."))
        .to("direct:route");

The headers `CamelFcrepoIdentifier` (FCREPO_IDENTIFIER) and `CamelFcrepoBaseUrl` (FCREPO_BASE_URL) will continue to be supported, but, instead routes should start making use of the `CamelFcrepoUri` (FCREPO_URI) header. That header will also work just like `HTTP_URI` with the `http4` component:

    setHeader(FCREPO_URI).constant("http://example.org/fcrepo/rest")
    .to("fcrepo:localhost");

this route will send the fedora request to http://example.org/fcrepo/rest, not http://localhost.

File: fcrepo-api-x-indexing/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -34,6 +33,8 @@
     <property name="reindexStream" value="${service.reindex.stream}" />
   </bean>
 
+  <bean id="http" class="org.apache.camel.component.http4.HttpComponent" />
```
Comment (acoburn) [inline fcrepo-api-x-indexing/src/main/resources/OSGI-INF/blueprint/blueprint.xml:36]: I tend to add this line, too: `<bean id="https" class="org.apache.camel.component.http4.HttpComponent" />`

This allows you to define the reindexing/triplestore URLs as https:// URLs, too.

**Comment by acoburn** [comment]
And if you're wondering why I would remove the JMS header class -- those headers are not and will not be part of the future Fedora API specification. Instead, the spec will concern itself only with the message body, which is what code should start depending on, since a non-JMS-based messaging system will certainly not contain `org.fcrepo.jms.*` headers.

Eventually, the message format will conform to the W3C Activity Streams 2.0 spec. At that point, existing code that follows the pattern above will need little to no changes. See here for some sample message serializations: http://fcrepo.github.io/fcrepo-specification/#message-examples

File: fcrepo-api-x-indexing/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 1 above)
Comment (birkland) [reply fcrepo-api-x-indexing/src/main/resources/OSGI-INF/blueprint/blueprint.xml:36]: Are there any subtle nuances to doing that, or does it Just Work™?

Comment (acoburn) [reply fcrepo-api-x-indexing/src/main/resources/OSGI-INF/blueprint/blueprint.xml:36]: In my experience, it "just works".

**Comment by birkland** [comment]
Ah, I see.  It looks like we'll need to be on fcrepo-camel 4.5 before we can use the new recommended approach.  How about we move towards the recommended FCREPO_URI pattern as part of #94 (that way we can test it in an IT), and leave the obsolete `JmsHeaders` in place until that happens?

**Comment by acoburn** [comment]
@birkland yes, you'll need to be on fcrepo-camel/4.5 before implementing this -- for now, keeping the jms headers in place makes sense.

**Comment by emetsger** [comment]
Merged 9bd1ebcf

**Pull request closed without merging**